api/scpca_portal/models/dataset.py

```diff
@@ -342,35 +339,31 @@ def original_files(self) -> Iterable[OriginalFile]:
     def original_file_paths(self) -> Set[Path]:
         return {Path(of.s3_key) for of in self.original_files}
 
-    @property
-    def metadata_file_map(self) -> Dict[str, str]:
-        metadata_file_map = {}
+    def get_metadata_file_content(self, libraries: Iterable[Library]) -> str:
+        libraries_metadata = Library.get_libraries_metadata(libraries)
+        return metadata_file.get_file_contents(libraries_metadata)
+
+    def get_project_modality_libraries(
+        self, project_id: str, modality: Modalities
+    ) -> Iterable[Library]:
+        return Library.objects.filter(samples__in=self.get_samples(project_id, modality)).distinct()
```

Comment:
I think we should also be filtering the libraries on the modality as well here. Otherwise we would be getting all libraries for each sample that matches on that project/sample combination.